cli/hooks/webpack.js

```diff
@@ -0,0 +1,165 @@
+// eslint-disable-next-line security/detect-child-process
+const { execSync } = require('child_process');
+const fs = require('fs-extra');
+const path = require('path');
+const semver = require('semver');
+const util = require('util');
+require('colors');
+
+const appcd = require('../lib/webpack/appcd');
+const WebpackService = require('../lib/webpack/service');
+
+const MIN_APPCD_VERSION = '3.2.0';
+
+const createWebpackLogger = (logger) => {
+	const webpackLogger = {
+		formatWithBadge(args) {
+			return `${' WEBPACK '.bgCyan.black} ${util.format(...args)}`;
+		}
+	};
+	[ 'info', 'debug', 'warn', 'error' ].forEach(level => {
+		webpackLogger[level] = (...args) => logger[level](webpackLogger.formatWithBadge(args));
+	});
+	return webpackLogger;
+};
+
+exports.id = 'ti.webpack';
+exports.init = (logger, config, cli) => {
+	let commandName;
+	let isWebpackEnabled = false;
+	let projectType;
+	let client;
+
+	cli.on('cli:command-loaded', (hookData) => {
+		const command = hookData.command;
+		commandName = command.name;
+	});
+
+	cli.on('cli:post-validate', () => {
+		projectType = getWebpackProjectType(cli.argv['project-dir']);
+		if (projectType === null) {
+			return;
+		}
+		isWebpackEnabled = true;
+		process.env.TI_USE_WEBPACK = true;
+
+		if (commandName !== 'build') {
+			return;
+		}
+
+		let appcdRootPath;
+		const isAppcCli = typeof process.env.APPC_ENV !== 'undefined';
+		if (isAppcCli) {
+			// we were invoked by appc-cli, load bundled daemon client.
+			appcdRootPath = resolveModuleRoot('appcd');
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find Appcelerator Daemon inside the appc-cli. Please make sure to use a recent appc-cli version. You can install/select it with  "appc use latest".');
+			}
+		} else {
+			// plain ti-cli, load daemon from global modules
+			const globalModulesPath = execSync('npm root -g').toString().replace(/\s+$/, '');
+			appcdRootPath = resolveModuleRoot('appcd', [ globalModulesPath ]);
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find global Appcelerator Daemon install. You can install it with "npm i appcd -g".');
+			}
+		}
+		ensureAppcdVersion(appcdRootPath, MIN_APPCD_VERSION, isAppcCli);
+		const AppcdClient = resolveAppcdClient(appcdRootPath);
+		const baseClient = new AppcdClient();
+		if (isAppcCli) {
+			baseClient.appcd = path.resolve(appcdRootPath, '..', '.bin', 'appcd');
```

Comment:
@janvennemann This will need to be `appcd.cmd` when running on Windows as the appcd binary will fail to start (just confirmed with @ssjsamir)